### PR TITLE
Fixed #25753 -- Made get_format() cache the formats from Django settings

### PR DESCRIPTION
**Pull request by claudep**
I took and polished Jaap's patch from #5655.

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
buildbot, test this please.

File: django/utils/formats.py [hunk 1]
```diff
@@ -112,32 +112,40 @@ def get_format(format_type, lang=None, use_l10n=None):
...
+        lang = get_language()
+    cache_key = (format_type, lang)
+    try:
+        return _format_cache[cache_key]
+    except KeyError:
+        pass
+
+    # The requested format_type has not been cached yet. Try to find it in any of the
+    # format_modules for the given lang if l10n is enabled. If it's not there, or l10n
+    # is disabled we fall back to the project settings.
```
Comment (timgraham) [inline django/utils/formats.py]: Check line length (79)
chop "we"
"If it's not there or if l10n is disabled, fall back.."

File: django/utils/formats.py [hunk 2]
```diff
@@ -112,32 +112,40 @@ def get_format(format_type, lang=None, use_l10n=None):
...
+                if val is not None:
+                    break
+            except AttributeError:
+                pass
+    if val is None:
+        if format_type not in FORMAT_SETTINGS:
+            return format_type
+        val = getattr(settings, format_type)
+    elif format_type in ISO_INPUT_FORMATS.keys():
+        # If we got a list of input formats from one of the format_modules we need to make sure
```
Comment (timgraham) [inline django/utils/formats.py]: `# If a list of input formats from one of the format_modules was retrieved, make sure...`

File: tests/i18n/tests.py [hunk 3]
```diff
@@ -731,17 +731,16 @@ def test_false_like_locale_formats(self):
         even if they would be interpreted as False in a conditional test
```
Comment (timgraham) [inline tests/i18n/tests.py]: Remove the unused `patch_formats` function.

Comment (claudep) [inline tests/i18n/tests.py]: It's still used in `test_sanitize_separators`, to force testing a potential `THOUSAND_SEPARATOR='.', DECIMAL_SEPARATOR=','` configuration.

**Comment by jaap3** [comment]
Thank you so much for picking up this patch!

**Pull request merged**